url.js

```diff
@@ -76,9 +76,12 @@
     var instance = URLUtils(url || '');
 
     // Detect for ES5 getter/setter support
-    var ES5_GET_SET = (Object.defineProperties && (function () {
-      var o = {}; Object.defineProperties(o, { p: { 'get': function () { return true; } } }); return o.p;
-    }()));
+    var ES5_GET_SET = false;
+    try {
+      ES5_GET_SET = (Object.defineProperties && (function () {
```

Comment:
And maybe for readability, we want to turn the whole thing into an IIFE rather than just the core, e.g.:
```js
var ES5_GET_SET = (function() {
  if (!('defineProperties' in Object)) return false;
  try {
    ...
  } catch (_) {}
}());
```